### PR TITLE
Fixing Larger Issues with previous PR

### PR DESCRIPTION
**Pull request by MikeLippincott**
Added .ipynb converted files for a better review.

### CODE REVIEW

File: 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Wave1 Exploratory Data Analysis on dilated 50 data
+
+# Here we understand more about the data and how it is distrubuted.
+
+# In[16]:
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py:14]: I will add this into the next PR, and into my practices

File: 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+# Import data with low memory arg as the data are large 
+df = pd.read_csv("../../Extracted Features (CSV files)/interstellar_wave1_dilate50_sc.csv.gz",low_memory=False)
+
+
+# In[18]:
+
+
+# Function to display df shape and # of replicates
+def df_stats(df):
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py:36]: I will also add this into the next commit as I am learning

File: 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Wave1 Exploratory Data Analysis on dilated 50 data
+
+# Here we understand more about the data and how it is distrubuted.
+Wave1 data are extracted features from raw images.
```
Comment (jenna-tomkinson) [inline 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py:7]: I think these should be commented not code.

File: 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jenna-tomkinson) [reply 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py:14]: Sounds good! This same comment also goes for the notebook, which starts at 25. So we want to always start from 1.

File: 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jenna-tomkinson) [reply 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py:36]: Looks like you moved this and other functions over into another file so I recommend removing these functions from the notebook and py file. 

As well, when you import the packages, you should import the utils as well (e.g., as an example from one of my projects, `import extraction_utils as utils`)  so then you can use them in the notebook.

File: 0.Exploratory_Data_Analysis/utils/utils.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  
+  ax= ax.ravel()
+
+  label = np.arange(n_clusters)
+  for i in label:
+    plot_barchart_all_unique_features(df, label_col, label=i, ax=ax[i], sensitivity=sensitivity)
+    ax[i].xaxis.set_tick_params(labelbottom=True)
+    ax[i].yaxis.set_tick_params(labelsize=4)
+
+  plt.tight_layout()
```
Comment (jenna-tomkinson) [inline 0.Exploratory_Data_Analysis/utils/utils.py:115]: Add newline at the end of this file.

**Review by jenna-tomkinson — APPROVED** [review]
I think this PR is good! Feel free to merge after you look at the further comments I made. 

I would recommend in PRs that you always respond to all comments, even if you don't plan on following the suggestion, just so that the reviewer sees what you plan to do or why you didn't use the suggestion. Plus, it is good for documentation, in general, to always respond to comments so that you can go back to these PRs and see all of the changes you did or didn't make.

File: 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+
+# In[22]:
+
+
+# Subset df where n = number of subset datam points 
+df_subset = df.sample(n=1500)
+
+# Define which columns are data and which are descriptive...
+df_descriptive = df_subset[["Metadata_wellName",
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py]: code snippet added! Thank you for the concise alternative!

File: 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+
+
+treatment_ids = df_descriptive['Metadata_treatment']
+
+
+# In[23]:
+
+
+# clustering code adapted from https://www.kaggle.com/code/aussie84/clustering-with-kmeans-pca-tsne
+import numpy as np # linear algebra
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py]: Fixed via pre-commit. Thank you @axiomcura

File: 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+def outside_limit(df, label_col, label, sensitivity):
+  feature_list = df.columns[:-1]
+  
+  plot_list = []
+  mean_overall_list = []
+  mean_cluster_list = []
+  
+  for i,varname in enumerate(feature_list):
+    
+    #     get overall mean for a variable, set lower and upper limit
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py]: Fixed.

File: 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+    mean_overall = df[varname].mean()
+    lower_limit = mean_overall - (mean_overall*sensitivity)
+    upper_limit = mean_overall + (mean_overall*sensitivity)
+
+    #     get cluster mean for a variable
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py]: also fixed

File: 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Exploratory_Data_Analysis/nbconverted/Wave1_eda_dialate50_sc.py]: Done, helper functions are in another file and imported to notebook

**Pull request merged**